File: app/src/main/java/com/swathi/queue_app/model/DoctorModel.kt
package com.swathi.queue_app.model

import com.google.gson.annotations.SerializedName

data class DoctorModel(

    @SerializedName("_id")
    val id: String,

    val hospitalcode: String,

    val doctorName: String,

    val specialization: String,

    val qualification: String,

    val roomNumber: String,

    val availableDays: List<String>,

    val startTime: String,

    val endTime: String,
    val createdAt: String,

    val updatedAt: String
)

File: Backend/models/doctor_model.js
const mongoose = require("mongoose");

const doctorSchema = new mongoose.Schema({

    hospitalcode: {
        type: String,
        ref: "Hospital",
        required: true
    },

    doctorName: {
        type: String,
        required: true
    },

    specialization: {
        type: String,
        required: true
    },

    qualification: {
        type: String
    },

    roomNumber: {
        type: String
    },

    availableDays: [{
        type: String
    }],

    startTime: {
        type: String
    },

    endTime: {
        type: String
    }

}, {
    timestamps: true
});

module.exports = mongoose.model("Doctor", doctorSchema);